SRDB/convert.py

```diff
@@ -0,0 +1,235 @@
+# functions in the order that they are used in main()
+#####################################################
+
+# 1. if not present in current dir, download the necessary data
+def download_rs_data(local_data, cwd):
+    
+    print("""
+    An Earthdata username and password is required to download this data. 
+    Create an account here: https://urs.earthdata.nasa.gov/users/new')\n"""
+         )
+    auth = earthaccess.login() # user must input username/password
+
+    # search for collections of this name; returns dict of collections
+    dname = 'Global Gridded 1-km Annual Soil Respiration and Uncertainty Derived from SRDB V3'
+    query = DataCollections().keyword(dname)
+    
+    # print a dictionary of collections with their names/abstracts
+    collections = query.fields(['ShortName','Abstract']).get()
+    collection = collections[0] # choose the collection
+    cid = collection.concept_id() # get collection cid
+    
+    # using the cid, get a list of granules (.tif files)
+    query = earthaccess.granule_query().concept_id(cid)
+    granules = query.get()
+    
+    # download the data (or don't if already present)
+    file1 = earthaccess.download(granules[3], cwd) # Rs .tif/.sha256
+    file2 = earthaccess.download(granules[4], cwd) # uncertainty .tif/.sha256
+    download_stamp = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(os.path.getmtime(local_data)))
+
+    return download_stamp
+
+# 2. create xarray dataset
+def create_xarray(local_data, local_u_data, uncertainty, cwd):
+
+    def open_raster(local_data, cwd):
+
+        # read a .tif file into an xarray
+        data = rxr.open_rasterio(local_data, band_as_variable=True)
+        epsg_code = int(data.rio.crs.to_epsg())
+        if epsg_code != 4326:
+            data = data.rio.reproject(crs=proj)
+        return data
+    
+    # open data and error files
+    data = open_raster(local_data, cwd)
+    errd = open_raster(local_u_data, cwd)
+    data = data.assign(uncertainty=errd['band_1'])
+    return data
+
+# 3. resample to 0.5 degrees
+def coarsen(data, target_res):
+    
+    resampled_data = data.coarsen(x=(int(target_res / abs(data.rio.resolution()[0]))),
+                                  y=(int(target_res / abs(data.rio.resolution()[1])))).mean()
+    return resampled_data
+
+# 4(a). subfunction to convert original units to target units
+def convert_units(data_array, target_units):
+    
+    # Get the original units from the data array attributes
+    original_units = data_array.attrs.get('units', None)
+    if original_units is None:
+        raise ValueError('Original units not found in data array attributes.')
+    
+    # Create unit objects for original and target units
+    original_units_obj = cfunits.Units(original_units)
+    target_units_obj = cfunits.Units(target_units)
+    
+    # Handle specific cases with intermediate conversion steps
+    if original_units == 'g m-2 y-1' and target_units == 'kg m-2 s-1':
```

Comment:
Yeah, it gave me an error, which is why I threw in the if-statement. I can give cf-units a try! I didn't realize there were two different ones... :-)

Reply:
At one point I had some PR's to them and they didn't respond and I made the switch. I wonder if it is because `y` is not a unit in the UDUNITS database. You may try `y-1` --> `yr-1` and then see if the package just converts the units?

Reply:
What do you think of:

```python
original_units = original_units.replace("y-1","yr-1")
```

and then your conform line?